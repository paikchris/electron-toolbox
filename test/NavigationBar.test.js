// Link.react.test.js
import React from "react";
import renderer from "react-test-renderer";
import * as Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { MemoryRouter } from "react-router-dom";
import { NavigationBar } from "./../src/components/NavigationBar";

//ENZYME SETUP
Enzyme.configure({ adapter: new Adapter() });

//TESTS
describe("Navigation Bar Tests", () => {
	test("NavBar Snapshot Test", () => {
		const tree = renderer
			.create(<MemoryRouter>
				<NavigationBar />
			</MemoryRouter>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	test("Enzyme Shallow Render Example", () => {
		const wrapper = Enzyme.shallow(<NavigationBar />);
	});
	test("Enzyme Full Render Example", () => {
		const wrapper = Enzyme.mount(<MemoryRouter>
			<NavigationBar />
		</MemoryRouter>);
	});
});
