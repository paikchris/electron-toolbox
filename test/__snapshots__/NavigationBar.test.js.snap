// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar Tests NavBar Snapshot Test 1`] = `
<div
  className="pt-navbar"
>
  <div
    className="pt-navbar-group pt-align-left"
  >
    <a
      className="rL"
      href="/"
      onClick={[Function]}
    >
      <div
        className="pt-navbar-heading"
      />
    </a>
  </div>
  <div
    className="pt-navbar-group pt-align-right"
  >
    <div
      className="pt-navbar-divider"
    />
    <a
      className="rL"
      href="/"
      onClick={[Function]}
    >
      <button
        className="pt-button pt-minimal"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <svg
          className="pt-icon"
          data-icon="home"
          height={16}
          style={Object {}}
          viewBox="0 0 16 16"
          width={16}
        >
          <title>
            home
          </title>
          <path
            d="M2 10v5c0 .55.45 1 1 1h3v-5h4v5h3c.55 0 1-.45 1-1v-5L8 4l-6 6zm13.71-2.71L14 5.59V2c0-.55-.45-1-1-1s-1 .45-1 1v1.59L8.71.29C8.53.11 8.28 0 8 0s-.53.11-.71.29l-7 7a1.003 1.003 0 0 0 1.42 1.42L8 2.41l6.29 6.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="pt-button-text"
        >
          Home
        </span>
      </button>
    </a>
    <a
      className="rL"
      href="/projects/index"
      onClick={[Function]}
    >
      <button
        className="pt-button pt-minimal"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <svg
          className="pt-icon"
          data-icon="projects"
          height={16}
          style={Object {}}
          viewBox="0 0 16 16"
          width={16}
        >
          <title>
            projects
          </title>
          <path
            d="M14 3c0-.55-.45-1-1-1H3c-.55 0-1 .45-1 1v1h12V3zm-2-3H4c-.55 0-1 .45-1 1h10c0-.55-.45-1-1-1zm3 5H1c-.55 0-1 .45-1 1v9c0 .55.45 1 1 1h14c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zm-3 6c0 .55-.45 1-1 1H5c-.55 0-1-.45-1-1V9h1v2h6V9h1v2z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="pt-button-text"
        >
          Projects
        </span>
      </button>
    </a>
    <div
      className="pt-navbar-divider"
    />
    <button
      className="pt-button pt-minimal"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <svg
        className="pt-icon"
        data-icon="user"
        height={16}
        style={Object {}}
        viewBox="0 0 16 16"
        width={16}
      >
        <title>
          user
        </title>
        <path
          d="M7.99-.01A7.998 7.998 0 0 0 .03 8.77c.01.09.03.18.04.28.02.15.04.31.07.47.02.11.05.22.08.34.03.13.06.26.1.38.04.12.08.25.12.37.04.11.08.21.12.32a6.583 6.583 0 0 0 .3.65c.07.14.14.27.22.4.04.07.08.13.12.2l.27.42.1.13a7.973 7.973 0 0 0 3.83 2.82c.03.01.05.02.07.03.37.12.75.22 1.14.29l.2.03c.39.06.79.1 1.2.1s.81-.04 1.2-.1l.2-.03c.39-.07.77-.16 1.14-.29.03-.01.05-.02.07-.03a8.037 8.037 0 0 0 3.83-2.82c.03-.04.06-.08.09-.13.1-.14.19-.28.28-.42.04-.07.08-.13.12-.2.08-.13.15-.27.22-.41.04-.08.08-.17.12-.26.06-.13.11-.26.17-.39.04-.1.08-.21.12-.32.04-.12.08-.24.12-.37.04-.13.07-.25.1-.38.03-.11.06-.22.08-.34.03-.16.05-.31.07-.47.01-.09.03-.18.04-.28.02-.26.04-.51.04-.78-.03-4.41-3.61-7.99-8.03-7.99zm0 14.4c-1.98 0-3.75-.9-4.92-2.31.67-.36 1.49-.66 2.14-.95 1.16-.52 1.04-.84 1.08-1.27.01-.06.01-.11.01-.17-.41-.36-.74-.86-.96-1.44v-.01c0-.01-.01-.02-.01-.02-.05-.13-.09-.26-.12-.39-.28-.05-.44-.35-.5-.63-.06-.11-.18-.38-.15-.69.04-.41.2-.59.38-.67v-.06c0-.51.05-1.24.14-1.72.02-.13.05-.26.09-.39.17-.59.53-1.12 1.01-1.49.49-.38 1.19-.59 1.82-.59.62 0 1.32.2 1.82.59.48.37.84.9 1.01 1.49.04.13.07.26.09.4.09.48.14 1.21.14 1.72v.07c.18.08.33.26.37.66.03.31-.1.58-.16.69-.06.27-.21.57-.48.62-.03.13-.07.26-.12.38 0 .01-.01.04-.01.04-.21.57-.54 1.06-.94 1.42 0 .06.01.13.01.19.04.43-.12.75 1.05 1.27.65.29 1.47.6 2.14.95a6.415 6.415 0 0 1-4.93 2.31z"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <button
      className="pt-button pt-minimal"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <svg
        className="pt-icon"
        data-icon="notifications"
        height={16}
        style={Object {}}
        viewBox="0 0 16 16"
        width={16}
      >
        <title>
          notifications
        </title>
        <path
          d="M8 16c1.1 0 2-.9 2-2H6c0 1.1.9 2 2 2zm6-5c-.55 0-1-.45-1-1V6c0-2.43-1.73-4.45-4.02-4.9 0-.04.02-.06.02-.1 0-.55-.45-1-1-1S7 .45 7 1c0 .04.02.06.02.1A4.992 4.992 0 0 0 3 6v4c0 .55-.45 1-1 1s-1 .45-1 1 .45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <button
      className="pt-button pt-minimal"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <svg
        className="pt-icon"
        data-icon="cog"
        height={16}
        style={Object {}}
        viewBox="0 0 16 16"
        width={16}
      >
        <title>
          cog
        </title>
        <path
          d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 0 0-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 0 0-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
